pkg/osquery/enrollment_details.go

```diff
@@ -145,3 +148,31 @@ func getOsqEnrollDetails(ctx context.Context, osquerydPath string, details *serv
 
 	return nil
 }
+
+// CollectAndSetEnrollmentDetails collects enrollment details from osquery and sets them in the knapsack.
+func CollectAndSetEnrollmentDetails(ctx context.Context, slogger *slog.Logger, k types.Knapsack, collectTimeout time.Duration, collectRetryInterval time.Duration) {
+	ctx, span := traces.StartSpan(ctx)
+	defer span.End()
+
+	// Get the runtime details
+	details := getRuntimeEnrollDetails()
+
+	// Set the osquery version and save everything to knapsack before attempting to get osquery enrollment details
+	k.SetEnrollmentDetails(details)
+
+	latestOsquerydPath := k.LatestOsquerydPath(ctx)
+
+	if latestOsquerydPath != "" {
+		if err := backoff.WaitFor(func() error {
+			err := getOsqEnrollDetails(ctx, latestOsquerydPath, &details)
+			if err != nil {
+				span.AddEvent("failed to get enrollment details")
+			}
+			return err
+		}, collectTimeout, collectRetryInterval); err != nil {
+			span.AddEvent(fmt.Sprintf("enrollment details collection failed: %v", err))
```

Comment:
Optional -- you can use `traces.SetError` to update the span with the error and error status -- `traces.SetError(span, fmt.Sprintf("enrollment details collection failed: %w", err))`